shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/Rules/TestChain.hs

```diff
@@ -190,47 +221,72 @@ checkWithdrawlBound SourceSignalTarget {source, signal, target} =
 
 -- | If we are not at an Epoch Boundary, then (Utxo + Deposits)
 -- increases by Withdrawals min Fees (for all transactions in a block)
-utxoDepositsIncreaseByFeesWithdrawals :: SourceSignalTarget (CHAIN C) -> Property
+utxoDepositsIncreaseByFeesWithdrawals ::
+  ( ChainProperty era,
+    HasField "txfee" (Core.TxBody era) Coin,
+    HasField "wdrls" (Core.TxBody era) (Wdrl era)
+  ) =>
+  SourceSignalTarget (CHAIN era) ->
+  Property
 utxoDepositsIncreaseByFeesWithdrawals SourceSignalTarget {source, signal, target} =
   circulation target <-> circulation source
     === withdrawals signal <-> txFees signal
   where
     circulation chainSt =
       let es = (nesEs . chainNes) chainSt
           (UTxOState {_utxo = u, _deposited = d}) = (_utxoState . esLState) es
-       in balance u <+> d
+       in Val.coin (balance u) <+> d
 
 -- | If we are not at an Epoch Boundary, then (Utxo + Deposits + Fees)
 -- increases by sum of withdrawals for all transactions in a block
-potsSumIncreaseWdrlsPerBlock :: SourceSignalTarget (CHAIN C) -> Property
+potsSumIncreaseWdrlsPerBlock ::
+  ( ChainProperty era,
+    HasField "wdrls" (Core.TxBody era) (Wdrl era)
+  ) =>
+  SourceSignalTarget (CHAIN era) ->
+  Property
 potsSumIncreaseWdrlsPerBlock SourceSignalTarget {source, signal, target} =
   potsSum target <-> potsSum source === withdrawals signal
   where
     potsSum chainSt =
       let (UTxOState {_utxo = u, _deposited = d, _fees = f}) =
             _utxoState . esLState . nesEs . chainNes $ chainSt
-       in balance u <+> d <+> f
+       in Val.coin (balance u) <+> d <+> f
 
 -- | If we are not at an Epoch Boundary, then (Utxo + Deposits + Fees)
 -- increases by sum of withdrawals in a transaction
-potsSumIncreaseWdrlsPerTx :: SourceSignalTarget (CHAIN C) -> Property
+potsSumIncreaseWdrlsPerTx ::
+  forall era.
+  ( ChainProperty era,
+    HasField "certs" (Core.TxBody era) (StrictSeq (DCert era)),
+    HasField "wdrls" (Core.TxBody era) (Wdrl era)
+  ) =>
+  SourceSignalTarget (CHAIN era) ->
+  Property
 potsSumIncreaseWdrlsPerTx SourceSignalTarget {source = chainSt, signal = block} =
   conjoin $
     map sumIncreaseWdrls $
       sourceSignalTargets ledgerTr
   where
     (_, ledgerTr) = ledgerTraceFromBlock chainSt block
+    sumIncreaseWdrls :: SourceSignalTarget (LEDGER era) -> Property
     sumIncreaseWdrls
       SourceSignalTarget
         { source = (UTxOState {_utxo = u, _deposited = d, _fees = f}, _),
           signal = tx,
           target = (UTxOState {_utxo = u', _deposited = d', _fees = f'}, _)
         } =
-        (balance u' <+> d' <+> f') <-> (balance u <+> d <+> f)
-          === fold (unWdrl . _wdrls $ _body @C tx)
+        (Val.coin (balance u') <+> d' <+> f') <-> (Val.coin (balance u) <+> d <+> f)
+          === fold (unWdrl (getField @"wdrls" (_body tx)))
 
```

Comment:
here too - should this add up all the assets, not just check that coins add up correctly?